src/core/packages/saved-objects/base-server-internal/src/saved_objects_type_registry.ts

```diff
@@ -17,6 +18,11 @@ import type { SavedObjectsType, ISavedObjectTypeRegistry } from '@kbn/core-saved
  */
 export class SavedObjectTypeRegistry implements ISavedObjectTypeRegistry {
   private readonly types = new Map<string, SavedObjectsType>();
+  private readonly legacyTypes: string[];
+
+  constructor(legacyTypes: string[] = REMOVED_TYPES) {
```

Comment:
NIT we can make this cleaner / more future proof:
* The constructor parameter can be an Object e.g. `SavedObjectTypeRegistryConfig { legacyTypes: string[] }`, which we can default to `= {}`. This way, if we add more parameters in the future, we will not be forced to pass any parameters that we don't want to customise.
* The SavedObjectTypeRegistry should not have any knowledge about the REMOVED_TYPES, aka it defaults to `[]`. We will pass it where we instantiate the registry:
  * `src/core/packages/saved-objects/server-internal/src/saved_objects_service.ts`
  * `src/core/server/integration_tests/saved_objects/migrations/kibana_migrator_test_kit.ts`

Reply:
I tend to agree with Gerard, although we don't have a consistent practice of how we instantiate classes, we decide what is best case-by-case. In this case a `kwargs` style `{ blockList }: { blockList?: string[] } = {}` might save some future refactoring and perhaps is the best to default approach. The cost is making instantiation generally a little more verbose, but arguably clearer...

FWIW an alternative is to use the shorthand "Parameter properties" TS feature ([docs](https://www.typescriptlang.org/docs/handbook/2/classes.html#parameter-properties)) to avoid the boilerplate `this.instanceValue = instanceValue` assignment.